packages/sanity/src/structure/components/requestPermissionDialog/RequestPermissionDialog.tsx

```diff
@@ -0,0 +1,178 @@
+    client
+      .request<AccessRequest | null>({
+        url: `/access/project/${projectId}/requests`,
+        method: 'post',
+        body: {note, requestUrl: window?.location.href, requestedRole, type: 'role'},
+      })
+      .then((request) => {
+        if (request) {
+          if (onRequestSubmitted) onRequestSubmitted()
+          toast.push({title: 'Edit access requested'})
+        }
+      })
+      .catch((err) => {
+        const statusCode = err?.response?.statusCode
+        const errMessage = err?.response?.body?.message
+        if (statusCode === 429) {
+          // User is over their cross-project request limit
+          setHasTooManyRequests(true)
+          setMsgError(errMessage)
+        }
+        if (statusCode === 409) {
+          // If we get a 409, user has been denied on this project or has a valid pending request
+          // valid pending request should be handled by GET request above
+          setHasBeenDenied(true)
+          setMsgError(errMessage)
+        } else {
+          toast.push({
+            title: 'There was a problem submitting your request.',
+            status: errMessage,
+          })
+        }
+      })
+      .finally(() => {
+        setIsSubmitting(false)
+      })
+    setIsSubmitting(false)
+  }
+
+  return (
+    <DialogProvider position={'fixed'} zOffset={zOffset.fullscreen}>
+      <Dialog
+        width={1}
+        id={dialogId}
+        header={t('request-permission-dialog.header.text')}
+        footer={{
+          cancelButton: {
+            onClick: onClose,
+            text: t('confirm-dialog.cancel-button.fallback-text'),
+          },
+          confirmButton: {
+            loading: isSubmitting,
+            disabled: hasTooManyRequests || hasBeenDenied,
+            text: t('request-permission-dialog.confirm-button.text'),
+            tone: 'primary',
+            onClick: onConfirm,
+          },
+        }}
+        onClose={onClose}
+        onClickOutside={onClose}
+      >
+        <DialogBody>
+          <Stack space={4}>
+            <Text>{t('request-permission-dialog.description.text')}</Text>
+            {hasTooManyRequests || hasBeenDenied ? (
+              <Card tone={'caution'} padding={3} radius={2} shadow={1}>
+                <Text size={1}>
+                  {hasTooManyRequests && (
+                    <>{msgError ?? t('request-permission-dialog.warning.limit-reached.text')}</>
+                  )}
+                  {hasBeenDenied && (
+                    <>{msgError ?? t('request-permission-dialog.warning.denied.text')}</>
+                  )}
+                </Text>
+              </Card>
+            ) : (
+              <Stack space={3} paddingBottom={0}>
+                <TextInput
+                  placeholder={t('request-permission-dialog.note-input.placeholder.text')}
+                  disabled={isSubmitting}
+                  onKeyDown={(e) => {
+                    if (e.key === 'Enter') onConfirm()
```

Comment:
It it possible to solve this using an html `form` instead?

Reply:
It would be but I think we'd have to add a hidden input and also add `preventDefault` to the submit handler. Happy to add if that's a common pattern in the codebase, but just for the single input I think this is cleaner and more maintainable.

Reply:
I couldn't find references for this pattern in the studio, but I think mainly because we don't have forms with submit actions. 
If you think it's not too disrupting for your changes I think this could be the beginning of a nicer pattern which follows the html specs.

Would be nice to try to wrap the whole dialog in the form and use the confirm button as the submit one. By passing  `footer.confirmButton.type = "submit"`

Reply:
Hey @pedrobonamin ! Looping back on this after some more investigation. The Dialog component portals itself and its children so wrapping it in a <form> means the form ends up empty in the DOM and wrapping just the DialogBody component means the confirmation button in the Dialog footer can't be used to submit the form. So we'd need a hidden element.

I totally hear you on the HTML standard though and would love to see sanity-ui support this more easily in the future, but for now I think it's more ergonomic to just have this onKeyDown handler to add this additional submit accessibility.